handlers/postList.go

```diff
@@ -0,0 +1,47 @@
+package handlers
+
+import (
+	"log"
+	"net/http"
+
+	"github.com/google/uuid"
+	"github.com/aws/aws-lambda-go/events"
+	"github.com/mount-joy/thelist-lambda/db"
+)
+
+type postList struct {
+	db db.DB
+}
+
+func newPostList() RouteHandler {
+	return &postList{
+		db: db.DynamoDB(),
+	}
+}
+
+func (g *postList) match(request events.APIGatewayProxyRequest) bool {
+	return request.Path == "/lists" && request.HTTPMethod == "POST"
+}
+
+func (g *postList) handle(request events.APIGatewayProxyRequest) (interface{}, int) {
+	log.Printf("Running %s", request.Path)
+	listID, err := g.createList(request.Path)
+
+	if err != nil {
+		log.Printf("Error: %s", err.Error())
+		return nil, http.StatusInternalServerError
+	}
+
+	list := map[string]string{"listID": listID}
+
+	return list, http.StatusOK
+}
+
+func (g *postList) createList(path string) (string, error) {
+	listID := generateListID(path)
+	return listID, g.db.CreateList(listID)
```

Comment:
CreateList takes a pointer to a string:

https://github.com/mount-joy/thelist-lambda/blob/81e77cce24a11e3ecacd1876907ff082bbc154c9/db/createList.go#L8

So this should be:

```suggestion
	return listID, g.db.CreateList(&listID)
```